src/sentry/replays/usecases/reader.py

```diff
@@ -238,9 +241,22 @@ def _fetch_segments_from_snuba(
 
 def download_segments(segments: List[RecordingSegmentStorageMeta]) -> Iterator[bytes]:
     """Download segment data from remote storage."""
+
+    # start a sentry transaction to pass to the thread pool workers
+    transaction = sentry_sdk.start_transaction(
```

Comment:
if you want these to be spans under the request transaction, just don't start this transaction here and then just do `with sentry_sdk.start_span(..)` below

Reply:
i assume the sentry span gets finished when the django endpoint returns (we start streaming the response) -- not sure what would happen if i did that, none of the current child spans end up in the endpoint transaction now e.g. https://sentry.sentry.io/performance/sentry:3a5b5275f5fb4206924a9f14b875c965/?project=1&query=http.method%3AGET&statsPeriod=14d&transaction=%2Fapi%2F0%2Fprojects%2F%7Borganization_slug%7D%2F%7Bproject_slug%7D%2Freplays%2F%7Breplay_id%7D%2Frecording-segments%2F

Reply:
hmm ok I've never tried streaming stuff while testing the SDK, but yea if nothing's waiting on the thread pool stuff to finish, the parent transaction has no way of ensuring timings, so then technically, a new transaction makes sense.

However, dynamic sampling, tracing etc is broken if you just call `start_transaction` right now, you'd need to do the whole [`continue_from_headers`](https://github.com/getsentry/sentry-python/blob/c147d7129f82845a69a2dba68b0e62598dbfe295/sentry_sdk/tracing.py#L271-L275) dance. This is something we're aware is broken and not user friendly (transaction within transactions within a single SDK).